src/main/java/frc/robot/subsystems/endgame/EndgameIOSim.java

```diff
@@ -1,25 +1,135 @@
+            profileTimer.reset();
+            profileTimer.start();
+
+            targetPosition = position;
+
+            initialPosition = elevatorSim.getPositionMeters();
+            initialVelocity = elevatorSim.getVelocityMetersPerSecond();
+
+            if (position < targetPosition) {
+                // Moving down, assume weight of robot is on the climber now
+                setFF(EndgameConstants.climberkFFRobot);
+            } else {
+                // Moving up, we only have to account for the weight of pushing the climber up
+                setFF(EndgameConstants.climberkFFClimber);
+            }
+        }
     }
 
+    // Were we moving down on last tick?
+    // This is used to keep track of whether or not we need to switch the elevatorSim object out.
+    boolean movingDownLast = false;
+
     @Override
     public void updateInputs(EndgameIOInputs inputs) {
         elevatorSim.update(Constants.loopTime);
-        elevatorSim.setInputVoltage(appliedVolts);
 
+        if (targetPosition < elevatorSim.getPositionMeters() && !movingDownLast) {
+            // We were not moving down before, but now we are, so we must update the sim to be
+            // simulating lifting the whole weight of the robot up
+            ElevatorSim newElevatorSim =
+                    new ElevatorSim(
+                            DCMotor.getNeoVortex(2),
+                            20,
+                            1.814
+                                    - EndgameConstants
+                                            .simRobotMass, // Add in negative robot mass, because the
+                            // robot will be pulling up on the elevator.
+                            0.02231009,
+                            0.0,
+                            0.45,
+                            true,
+                            0.0);
+            newElevatorSim.setState(
+                    VecBuilder.fill(
+                            elevatorSim.getPositionMeters(),
+                            elevatorSim.getVelocityMetersPerSecond()));
+            elevatorSim = newElevatorSim;
+
+            // Update state of direction we're currently moving
+            movingDownLast = true;
+        } else if (movingDownLast) {
+            // We were moving down before, but now we are not. Create a new simulator without the
+            // weight from lifting the robot.
+            ElevatorSim newElevatorSim =
+                    new ElevatorSim(
+                            DCMotor.getNeoVortex(2), 20, 1.814, 0.02231009, 0.0, 0.45, true, 0.0);
+            newElevatorSim.setState(
+                    VecBuilder.fill(
+                            elevatorSim.getPositionMeters(),
+                            elevatorSim.getVelocityMetersPerSecond()));
+            elevatorSim = newElevatorSim;
+
+            // Update state of direction we're currently moving
+            movingDownLast = false;
+        }
+
+        double appliedVolts;
```

Comment:
Coding best practice - can we initialize this to some value (0.0 would be fine) when it is declared?